devops/makeAPIGateway/makeAPIGateway.sh

```diff
@@ -0,0 +1,60 @@
+# !/bin/sh
+INITIALS_PROMPT="Please enter your first and last initial (or 'q' to quit)"
+NUM=2
+
+while :
+  do
+    # get and validate user initials
+    printf "%s\n" "$INITIALS_PROMPT" 
+    while read initials && [ "$initials" != "q" ];
+      do
+        # checks to see if string is null or has zero length
+        if [ -z "$initials" ]; then
+          printf '%s\n' "Error: initials can't be empty" 
+        # checks to see if the string matches the requested pattern
+        elif [[ "${initials}" =~ [^a-zA-Z] ]]; then
+          printf '%s\n' "Error: initials must only contain letters, a-z"
+        elif [[ "${#initials}" != $NUM ]]; then
+          printf '%s\n' "Error: initials must be $NUM characters"
+        else
+          break
+        fi
+          printf "%s\n" "$INITIALS_PROMPT" 
+      done
+
+    if [ "$initials" == "q" ]; then
+      printf '%s\n' "Done"
+      break
+    fi
+
+
+    # converts initials to lowercase and remove spaces
+    formattedInitials="$(echo "${initials}" | tr -d '[:space:]' | tr '[:upper:]' '[:lower:]')"
```

Comment:
Heads up: if this script ends up being ran through a github action, this way of making a random string wont work. You can instead use `openssl`:

```bash
OPENSSL_STRING=$(openssl rand -hex 3)
RANDOM_STRING=${OPENSSL_STRING:0:5}
```

See the file in .github/scripts in #69 for an example

Reply:
Thanks for the feedback! made these changes so there shouldn't be problems in the future